source/magpie/dataframe.d

```diff
@@ -73,6 +73,31 @@ private:
         alias aggregateType = Resolved;
     }
 
+    auto dropperRuntimeInternal(int[] positions)
+    {
+        import magpie.helper: dropper;
+        import std.algorithm: max, min, reduce;
+        assert(positions.reduce!min > -1 && positions.reduce!max < rows, "Index out of bound");
+
+        DataFrame!(true, RowType) ret;
+        ret.indx = Index();
+        ret.indx.indexing[1] = indx.indexing[1];
+        ret.indx.row.titles = indx.row.titles;
+
+        import std.range: lockstep;
+        foreach(a, b; lockstep(indx.row.index, indx.row.codes))
+        {
+            ret.indx.row.index ~= a;
+            ret.indx.row.codes ~= dropper(positions, b);
```

Comment:
in general try to avoid doing allocations in a loop, in particular this can be refactored to 
```
ret.indx.row.index = indx.row.index.dup;
ret.indx.row.codes = indx.row.codes.map!((e) => dropper(positions, e)).array.
```

Reply:
Sorry about that :sweat_smile:
Addressed in 3e3d0bb